python/ray/serve/application.py

```diff
@@ -0,0 +1,376 @@
+from typing import List, Dict, TextIO, Optional, Union
+import json
+import yaml
+import time
+import sys
+
+import ray
+from ray import serve
+from ray.serve.api import (
+    Deployment,
+    _get_global_client,
+)
+from ray.serve.common import DeploymentStatus
+from ray.dashboard.modules.serve.schema import (
+    ServeApplicationSchema,
+    serve_application_to_schema,
+    schema_to_serve_application,
+    serve_application_status_to_schema,
+)
+from ray.dashboard.modules.serve.sdk import ServeSubmissionClient
+from ray.serve.utils import logger
+
+
+class Application:
+    """
+    Contains a group of Serve deployments and manages their deployment to the
+    Ray cluster.
+
+    Args:
+        deployments (List[Deployment]): The Serve Application is
+            initialized to contain this group of deployments.
+    """
+
+    def __init__(self, deployments: List[Deployment] = None):
+        deployments = deployments or []
+
+        self._deployments = dict()
+        for idx, d in enumerate(deployments):
+            if isinstance(d, Deployment):
+                self._deployments[d.name] = d
```

Comment:
let's enforce uniqueness here

Reply:
might as well de-dup the code with `add_deployment` (just call it repeatedly here)